File: app/schema/actor.py
from typing import Optional, List

from pydantic import BaseModel, Field

from app.schema.home import SiteVideo
from app.schema.r import Page
from app.schema.video import SourceRef


class ImageInfo(BaseModel):
    width: Optional[int] = None
    height: Optional[int] = None
    mime: Optional[str] = None


class Actor(BaseModel):
    code: Optional[str] = None
    name: Optional[str] = None
    thumb: Optional[str] = None
    thumb_info: Optional[ImageInfo] = None
    alias: List[str] = Field(default_factory=list)

    source: SourceRef


class ActorPage(BaseModel):
    actor: Actor
    page: Page[list[SiteVideo]]
    is_favorite: bool = False


class ActorFavoriteCreate(BaseModel):
    site_id: int
    actor_code: str
    actor_name: Optional[str] = None
    actor_thumb: Optional[str] = None
    actor_alias: List[str] = Field(default_factory=list)


class ActorFavorite(ActorFavoriteCreate):
    id: int
    actor: Actor

File: frontend/src/types/actor.ts
import type {PagedResponse, SiteVideo, SourceRef} from "./video.ts";

export interface Actor {
    code?: string;
    name?: string;
    thumb?: string;
    alias: string[];
    source: SourceRef;
}

export interface ActorPageData {
    actor: Actor;
    page: PagedResponse<SiteVideo[]>;
    is_favorite: boolean;
}

export interface ActorFavoriteCreate {
    site_id: number;
    actor_code: string;
    actor_name?: string;
    actor_thumb?: string;
    actor_alias: string[];
}

export interface ActorFavorite extends ActorFavoriteCreate {
    id: number;
    actor: Actor;
}
